jerry-core/ecma/builtin-objects/ecma-builtin-date-prototype.cpp

```diff
@@ -38,6 +42,55 @@
  */
 
 /**
+ * Insert leading zeros to a string of a number if needed.
+ */
+static void
+ecma_date_insert_leading_zeros (ecma_string_t **str_p, /**< input/output string */
+                                ecma_number_t num, /**< input number */
+                                int32_t div) /**< "length" {10, 100, 1000} */
+{
+  JERRY_ASSERT (div == 10 || div == 100 || div == 1000);
+
+  while (num < div && div > 1)
+  {
+    div /= 10;
+
+    ecma_string_t *zero_str_p = ecma_new_ecma_string_from_number (ECMA_NUMBER_ZERO);
+    ecma_string_t *concat_p = ecma_concat_ecma_strings (zero_str_p, *str_p);
+    ecma_deref_ecma_string (zero_str_p);
+    ecma_deref_ecma_string (*str_p);
+    *str_p = concat_p;
+  }
+
+} /* ecma_date_insert_leading_zeros */
+
+/**
+ * Insert a number to the start of a string with a specific separator character and
+ * fix length.
```

Comment:
I could argue with that, but I added more more a more descriptive comment for this function. :)